File: MasterDirectory/MasterDirectory.Web/Modules/Inmobiliaria/CatalogosInmobiliaria/CatalogosInmobiliariaForm.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel.DataAnnotations;

namespace MasterDirectory.Inmobiliaria.Forms;

[FormScript("Inmobiliaria.CatalogosInmobiliaria")]
[BasedOnRow(typeof(CatalogosInmobiliariaRow), CheckNames = true)]
public class CatalogosInmobiliariaForm
{
    [Editable(false)]
    public int IdtipoCatalogo { get; set; }
    [Editable(false)]
    public int IdClave { get; set; }
    public string Descripcion { get; set; }
    [BooleanEditor]
    public int Activo { get; set; }
    //public DateTime DtRegistro { get; set; }
}

File: MasterDirectory/MasterDirectory.Web/Modules/ServerTypes/Inmobiliaria/CatalogosInmobiliariaForm.ts
import { LookupEditor, IntegerEditor, StringEditor, BooleanEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface CatalogosInmobiliariaForm {
    IdtipoCatalogo: LookupEditor;
    IdClave: IntegerEditor;
    Descripcion: StringEditor;
    Activo: BooleanEditor;
}

export class CatalogosInmobiliariaForm extends PrefixedContext {
    static formKey = 'Inmobiliaria.CatalogosInmobiliaria';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CatalogosInmobiliariaForm.init)  {
            CatalogosInmobiliariaForm.init = true;

            var w0 = LookupEditor;
            var w1 = IntegerEditor;
            var w2 = StringEditor;
            var w3 = BooleanEditor;

            initFormType(CatalogosInmobiliariaForm, [
                'IdtipoCatalogo', w0,
                'IdClave', w1,
                'Descripcion', w2,
                'Activo', w3
            ]);
        }
    }
}